src/status_im2/subs/chat/messages.cljs

```diff
@@ -112,27 +112,37 @@
-                                    {:album           (get albums album-id)
-                                     :album-id        album-id
-                                     :albumize?       (:albumize? message)
-                                     :message-id      (:message-id message)
-                                     :deleted?        (:deleted? message)
-                                     :deleted-for-me? (:deleted-for-me? message)
-                                     :deleted-by      (:deleted-by message)
-                                     :from            (:from message)
-                                     :timestamp-str   (:timestamp-str message)
-                                     :content-type    constants/content-type-album})
-                              (conj messages message))]
-               {:messages messages
-                :albums   albums}))
-           {:messages []
-            :albums   {}}
-           messages)
+   (reduce
+    (fn [{:keys [messages albums]} message]
+      (let [{:keys [album-id content quoted-message]} message
+            {:keys [response-to]}                     content
+            albums                                    (cond-> albums
+                                                        album-id (update album-id conj message))
+            messages                                  (if album-id
+                                                        (conj
+                                                         (filterv #(not= album-id (:album-id %))
+                                                                  messages)
+                                                         (cond->
+                                                           {:album (get albums album-id)
+                                                            :album-id album-id
+                                                            :albumize? (:albumize? message)
+                                                            :message-id (:message-id message)
+                                                            :deleted? (:deleted? message)
+                                                            :deleted-for-me? (:deleted-for-me? message)
+                                                            :deleted-by (:deleted-by message)
+                                                            :from (:from message)
+                                                            :timestamp-str (:timestamp-str message)
+                                                            :content-type constants/content-type-album}
+                                                           response-to
+                                                           (assoc-in [:content :response-to] response-to)
+                                                           quoted-message
+                                                           (assoc :quoted-message quoted-message)))
```

Comment:
why `assoc`? You can add them to the map like the rest of the keys right?

Reply:
fixed